src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -247,6 +247,83 @@
+ *                  break;
+ *          }
+ *      }
+ *      return trim( "# {$title}\n\n{$text_content}\n" );
+ *
+ * ### Tokens and _modifiable text_.
+ *
+ * #### Special "atomic" HTML elements.
+ *
+ * Not all HTML elements are able to contain other elements inside of them.
+ * For instance, the contents inside a TITLE element are plaintext (except
+ * that character references like &amp; will be decoded). This means that
+ * if the string `<img>` appears inside a TITLE element, then it's not an
+ * image tag, but rather it's text describing an image tag. Likewise, the
+ * contents of a SCRIPT or STYLE element are handled entirely separately in
+ * a browser than the contents of other elements because they represent a
+ * different language than HTML.
+ *
+ * For these elements the Tag Processor treats the entire sequence as one,
+ * from the opening tag, including its contents, through its closing tag.
+ * This means that the it's not possible to match the closing tag for a
+ * SCRIPT element unless it's unexpected; the Tag Processor already matched
+ * it when it found the opening tag.
+ *
+ * The inner contents of these elements are that element's _modifiable text_.
+ *
+ * The special elements are:
+ *  - `SCRIPT` whose contents are treated as raw plaintext but supports a legacy
+ *    style of including Javascript inside of HTML comments to avoid accidentally
+ *    closing the SCRIPT from inside a Javascript string. E.g. `console.log( '</script>' )`.
+ *  - `TITLE` and `TEXTAREA` whose contents are treated as plaintext and then any
+ *    character references are decoded. E.g. "1 &amp;lt; 2 < 3" becomes "1 < 2 < 3".
+ *  - `IFRAME`, `NOSCRIPT`, `NOEMBED`, `NOFRAME`, `STYLE` whose contents are treated as
+ *    raw plaintext and left as-is. E.g. "1 &amp;lt; 2 < 3" remains "1 &amp;lt; 2 < 3".
+ *
+ * #### Other tokens with modifiable text.
+ *
+ * There are also non-elements which are atomic in nature and contain modifiable text.
+ *
+ *  - `#text` nodes, whose entire token _is_ the modifiable text.
+ *  - Comment nodes and nodes that became comments because of some syntax error. The
+ *    text for these nodes is the portion of the comment inside of the syntax. E.g. for
+ *    `<!-- comment -->` the text is `" comment "` (note that the spaces are part of it).
+ *  - `CDATA` sections, whose text is the content inside of the section itself. E.g. for
+ *    `<![CDATA[some content]]>` the text is `"some content"`.
+ *  - "Funky comments," which are a special case of invalid closing tags whose name is
+ *    invalid. The text for these nodes is the text that a browser would transform into
+ *    an HTML when parsing. E.g. for `</%post_author>` the text is `%post_author`.
+ *
+ * And there are non-elements which are atomic in nature but have no modifiable text.
+ *  - `DOCTYPE` nodes like `<DOCTYPE html>` which have no closing tag.
+ *  - The empty end tag `</>` which is ignored in the browser and DOM but exposed
+ *    to the HTML API.
```

Comment:
@sirreal I'm not sure yet on what do to here. can we tag this for follow-up after merge?

I'm a bit concerned about using specific property names here because this is supposed to be the explanatory section of the documentation and I don't want to couple the description to our own terms; I want it to read comfortable for someone coming in with an HTML background - that is, leave things a bit loose here to guide an understanding without pinning it to one specific technicality.

nonetheless I've taken another pass at the comment to update it based on how this has developed.